### PR TITLE
Use tags more

### PR DESCRIPTION
**Pull request by tobias-**
Use tags more by allowing filtering on only certain tags for event streams

### CODE REVIEW

File: tachikoma-grpc/src/main/kotlin/com/sourceforgery/tachikoma/tracking/DeliveryNotificationService.kt [hunk 1]
```diff
@@ -52,10 +53,13 @@ internal class DeliveryNotificationService(override val di: DI) : DIAware {
...
         if (emailData == null) {
             LOGGER.error("Got event with non-existing email " + deliveryNotificationMessage.emailMessageId)
-            null
-        } else {
-            deliveryNotificationMessage.toEmailNotification(emailData, request)
+            return@mapNotNull null
+        }
+        if (includeTags.isNotEmpty()) {
+            emailData.transaction.tags.any { it in includeTags }
         }
```
Comment (pgilmore) [inline tachikoma-grpc/src/main/kotlin/com/sourceforgery/tachikoma/tracking/DeliveryNotificationService.kt:60]: This does nothing?

**Review by pgilmore — CHANGES_REQUESTED** [review]

File: tachikoma-grpc/src/main/kotlin/com/sourceforgery/tachikoma/tracking/DeliveryNotificationService.kt (hunk 1 above)
Comment (spacecowboy) [reply tachikoma-grpc/src/main/kotlin/com/sourceforgery/tachikoma/tracking/DeliveryNotificationService.kt:60]: Missing `filter` ?

File: tachikoma-it/src/test/kotlin/com/sourceforgery/tachikoma/database/dao/EmailStatusEventDAOTest.kt [hunk 2]
```diff
@@ -105,4 +105,24 @@ class EmailStatusEventDAOTest : DIAware {
 
         assertEquals(0, emailStatusEvents.size)
     }
+
+    @Test
+    fun `find entries based on tags`() {
+        val authentication1 = daoHelper.createAuthentication("example.org")
+        daoHelper.createEmailStatusEvent(
```
Comment (spacecowboy) [inline tachikoma-it/src/test/kotlin/com/sourceforgery/tachikoma/database/dao/EmailStatusEventDAOTest.kt:112]: You should create a second event which does not have tag `B` I think

**Review by spacecowboy — CHANGES_REQUESTED** [review]

File: tachikoma-grpc/src/main/kotlin/com/sourceforgery/tachikoma/tracking/DeliveryNotificationService.kt (hunk 1 above)
Comment (tobias-) [reply tachikoma-grpc/src/main/kotlin/com/sourceforgery/tachikoma/tracking/DeliveryNotificationService.kt:60]: Fixed

File: tachikoma-it/src/test/kotlin/com/sourceforgery/tachikoma/database/dao/EmailStatusEventDAOTest.kt (hunk 2 above)
Comment (tobias-) [reply tachikoma-it/src/test/kotlin/com/sourceforgery/tachikoma/database/dao/EmailStatusEventDAOTest.kt:112]: Done

**Review by pgilmore — APPROVED** [review]

**Pull request merged**